pkg/history_drivers/memory_store/store.go

```diff
@@ -0,0 +1,26 @@
+package memory_store
+
+import (
+	"sync"
+
+	"github.com/inngest/inngest/pkg/execution/history"
+	"github.com/inngest/inngest/pkg/history_reader"
+	"github.com/oklog/ulid/v2"
+)
+
+var (
+	Singleton = &RunStore{
+		Data: map[ulid.ULID]RunData{},
+		Mu:   &sync.Mutex{},
+	}
+)
+
+type RunData struct {
+	Run     history_reader.Run
+	History []history.History
+}
+
+type RunStore struct {
+	Data map[ulid.ULID]RunData
+	Mu   *sync.Mutex
```

Comment:
Nit:  this could be a *sync.RWMutex, and the Writer could `Lock` while the Reader could `RLock`.  This effectively allows unlimited concurrent reads, with single-threaded writes.